### PR TITLE
Saas 13.3 hr expense fix improvement avd

### PR DESCRIPTION
**Pull request by Aurelienvd**
Description of the issue/feature this PR addresses: Small fixes and improvements for the hr_expense module. See Task's pad for information about the improvements.

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

Task ID: 2229597

### CODE REVIEW

File: addons/hr_expense/models/hr_expense.py [hunk 1]
```diff
@@ -161,7 +160,8 @@ def _onchange_product_id(self):
 
     @api.onchange('company_id')
     def _onchange_expense_company_id(self):
-        self.employee_id = self.env['hr.employee'].search([('user_id', '=', self.env.uid), ('company_id', '=', self.company_id.id)])
+        if not self._context.get("default_employee_id"):
```
Comment (kba-odoo) [inline addons/hr_expense/models/hr_expense.py:163]: 2 small cosmetic notes here:
  - `self._context` is from the old API, instead use `self.env.context`
  - prefer simple quotes as double quotes, to be more consistent with the rest

File: addons/hr_expense/models/hr_expense.py [hunk 2]
```diff
@@ -509,16 +509,15 @@ def refuse_expense(self, reason):
         self.sheet_id.message_post_with_view('hr_expense.hr_expense_template_refuse_reason',
                                              values={'reason': reason, 'is_sheet': False, 'name': self.name})
 
+    @api.model
+    def get_default_dashbord(self, expense_state):
```
Comment (kba-odoo) [inline addons/hr_expense/models/hr_expense.py:513]: You can correct the typo here: dashbord -> dashboard

File: addons/hr_expense/models/hr_expense.py [hunk 3]
```diff
@@ -523,15 +523,14 @@ def refuse_expense(self, reason):
...
-    def get_expense_dashbord(self):
-        if not self.env.user.employee_ids:
-            return
-        expenses = self.read_group(
-            [
-                ('employee_id', 'in', self.env.user.employee_ids.ids),
-                ('payment_mode', '=', 'own_account'),
-                ('state', 'in', ['draft', 'reported', 'approved'])
-            ], ['total_amount', 'currency_id', 'state'], ['state', 'currency_id'], lazy=False)
+    def get_default_dashboard(self, expense_state):
```
Comment (tivisse) [inline addons/hr_expense/models/hr_expense.py:526]: You can't change a method signature in a stable release ;)

See: https://github.com/odoo/odoo/wiki/Contributing#what-does-stable-mean

Comment (tivisse) [reply addons/hr_expense/models/hr_expense.py:526]: Already fixed btw.

File: addons/hr_expense/static/src/js/expense_views.js [hunk 4]
```diff
@@ -79,7 +82,38 @@ odoo.define('hr_expense.expenses.tree', function (require) {
...
+        /**
+         * @override
+         * This function copies BasicController._notifyInvalidFields instead of calling it through
+         * this._super. This is done to inject the employee_string inside the do_warn notification container instead
+         * of creating a new notification.
+         */
+
+         _notifyInvalidFields: function(invalidFields) {
+            var record = this.model.get(this.handle, {raw: true});
+            const employee_invisible = this.$el.find('.o_expense_form_employee_id')[0].classList.contains('o_invisible_modifier');
```
Comment (tivisse) [inline addons/hr_expense/static/src/js/expense_views.js:95]: This is quite overkill. Why not raising a classic UserError ? 

1/ This is standard
2/ This doesn't require to update the module (we're on a stable release here).

File: addons/hr_expense/static/src/js/expense_views.js [hunk 5]
```diff
@@ -79,7 +82,38 @@ odoo.define('hr_expense.expenses.tree', function (require) {
...
+            var fields = record.fields;
+            var warnings = invalidFields.map(function (fieldName) {
+                var fieldStr = fields[fieldName].string + ((cond(fieldName)) ? employee_string : "");
+                return _.str.sprintf('<li>%s</li>', _.escape(fieldStr));
+            });
+            warnings.unshift('<ul>');
+            warnings.push('</ul>');
+            this.do_warn(_t("The following fields are invalid:"), warnings.join(''));
+            }
+    })
```
Comment (tivisse) [inline addons/hr_expense/static/src/js/expense_views.js:107]: Missing semicolon.

File: addons/hr_expense/static/src/js/expense_views.js [hunk 6]
```diff
@@ -79,7 +82,38 @@ odoo.define('hr_expense.expenses.tree', function (require) {
...
+            const employee_string = ": The current user has no related employee. Please, create one.";
+            const cond = fieldName => (fieldName == 'employee_id') && !record.data.employee_id && employee_invisible;
+            var fields = record.fields;
+            var warnings = invalidFields.map(function (fieldName) {
+                var fieldStr = fields[fieldName].string + ((cond(fieldName)) ? employee_string : "");
+                return _.str.sprintf('<li>%s</li>', _.escape(fieldStr));
+            });
+            warnings.unshift('<ul>');
+            warnings.push('</ul>');
+            this.do_warn(_t("The following fields are invalid:"), warnings.join(''));
```
Comment (tivisse) [inline addons/hr_expense/static/src/js/expense_views.js:105]: New source terms in stable => Adapt the .pot files ;)

File: addons/hr_expense/views/hr_expense_views.xml [hunk 7]
```diff
@@ -524,15 +530,15 @@
                         <group name="employee_details">
                             <field name="employee_id" context="{'default_company_id': company_id}"/>
                             <field name="payment_mode"/>
-                            <field name="user_id" domain="[('share', '=', False)]"/>
```
Comment (tivisse) [inline addons/hr_expense/views/hr_expense_views.xml:527]: This goes against https://github.com/odoo-dev/odoo/commit/e428d473c39bbc6fc786ebecb3b7703953579a59

I suppose that this is a rebase issue, as I don't see this point on the spec.

**Comment by tivisse** [comment]
@robodoo r+

**Pull request closed without merging**